src/nbl/ext/RadixSort/RadixSort.cpp

```diff
@@ -0,0 +1,76 @@
+#include "nbl/ext/RadixSort/RadixSort.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace RadixSort
+{
+
+RadixSort::RadixSort(video::IDriver* driver, const uint32_t wg_size) : m_wg_size(wg_size)
+{
+	assert(nbl::core::isPoT(m_wg_size));
+
+	const asset::SPushConstantRange pc_range = { asset::ISpecializedShader::ESS_COMPUTE, 0u, sizeof(nbl_glsl_ext_RadixSort_Parameters_t) };
+
+	{
+		const uint32_t count = 2u;
+		video::IGPUDescriptorSetLayout::SBinding binding[count];
+		for (uint32_t i = 0; i < count; ++i)
+			binding[i] = { i, asset::EDT_STORAGE_BUFFER, 1u, video::IGPUSpecializedShader::ESS_COMPUTE, nullptr };
+
+		m_histogram_ds_layout = driver->createGPUDescriptorSetLayout(binding, binding + count);
+		m_histogram_pipeline_layout = driver->createGPUPipelineLayout(&pc_range, &pc_range + 1, core::smart_refctd_ptr(m_histogram_ds_layout));
+
+		m_histogram_pipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(m_histogram_pipeline_layout),
+			createShader("nbl/builtin/glsl/ext/RadixSort/default_histogram.comp", driver));
+	}
+
+	{
+		const uint32_t count = 3u;
+		video::IGPUDescriptorSetLayout::SBinding binding[count];
+		for (uint32_t i = 0; i < count; ++i)
+			binding[i] = { i, asset::EDT_STORAGE_BUFFER, 1u, video::IGPUSpecializedShader::ESS_COMPUTE, nullptr };
+
+		m_scatter_ds_layout = driver->createGPUDescriptorSetLayout(binding, binding + count);
+		m_scatter_pipeline_layout = driver->createGPUPipelineLayout(&pc_range, &pc_range + 1, core::smart_refctd_ptr(m_scatter_ds_layout));
+
+		m_scatter_pipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(m_scatter_pipeline_layout),
+			createShader("nbl/builtin/glsl/ext/RadixSort/default_scatter.comp", driver));
+	}
```

Comment:
wonder if you could do it so that all 4 shaders (histogram, upsweep, downsweep, scatter) bind their inputs/ouputs to the same default bindings (i.e. histogram to binding 0, initial radix sort array binding 1, scatter radix sort array binding 2) then you'd only need one default descriptor layout, and hence only one descriptor set per-pass (actually 2, one for even and one for odd passes).

Reply:
I'm running experiments for this.

Do you want me to create 1 ds layout for all the pipelines in this specific example or change the default ds layouts given by individual components? For example, `ext::Scan` has one ds layout for both upsweep and downsweep which returned by `ext::Scan::getDefaultDescriptorSetLayout` --do you want me to modify this layout?
In my opinion, `ext::Scan` is an individual component and it would be kinda weird to have its ds layout adjusted because of other extension. (`ext::RadixSort` here)

`ext::RadixSort` have 2 default layout I could try to merge them into one, if its worth the effort?

Reply:
> Do you want me to create 1 ds layout for all the pipelines in this specific example or change the default ds layouts given by individual components? 

The default, its already known that RadixSort cannot function without Scan, so changing the default set of radix sort to be a superset and match the one of Scan (or even use 2 sets, one for SSBO common to Scan and Radix and one just for Radix)

>  For example, ext::Scan has one ds layout for both upsweep and downsweep which returned by ext::Scan::getDefaultDescriptorSetLayout --do you want me to modify this layout?

No, using two sets would solve this.